### PR TITLE
feat: Update terra htlc

### PR DESCRIPTION
**Pull request by AndonMitev**
Changelog:
1. Update smart contract to check if msg.value is equal to swapParam.value, if there is any difference transaction will be reverted and htlc won't be created
2. Add additional check to validate if transactions are coming from our smart contract template and if they are not we are rejecting the verify

[Testnet contract deployment](https://finder.terra.money/bombay-12/tx/ACB549321D32D7A0C34A3428289AB80FC86FBD197B688DCC70DD65D785E13BA2)

[Mainnet contract deployment](https://finder.terra.money/columbus-5/tx/D12EA939287C1AEF10086BAE26BC091D7D3B718FD7132A5915420B1FF9E553B0)

### CODE REVIEW

File: packages/terra-swap-provider/contract/src/contract.rs [hunk 1]
```diff
@@ -169,4 +169,4 @@ pub fn deduct_tax(deps: Deps, coin: Coin) -> StdResult<Coin> {
         denom: coin.denom,
         amount: (Uint256::from(coin.amount) - tax_amount).into(),
     })
-}
+}
```
Comment (kraikov) [inline packages/terra-swap-provider/contract/src/contract.rs]: Add new line please.

File: packages/terra-swap-provider/contract/src/error.rs [hunk 2]
```diff
@@ -14,6 +14,9 @@ pub enum ContractError {
     NoBalance {},
     #[error("Query not implemented")]
     QueryNotImplemented {},
+    #[error("Invalid amount of coins")]
+    InvalidAmountOfCoins {},
+    
     // Add any other custom errors you like here.
     // Look at https://docs.rs/thiserror/1.0.21/thiserror/ for details.
 }
```
Comment (kraikov) [inline packages/terra-swap-provider/contract/src/error.rs]: Add new line please.

**Review by kraikov — CHANGES_REQUESTED** [review]

**Review by kraikov — APPROVED** [review]
LGTM

**Review by kraikov — CHANGES_REQUESTED** [review]
Add `codeId` check in `verifyInitiateSwapTransaction` for both testnet and mainnet

**Review by harshjv — APPROVED** [review]

**Review by kraikov — APPROVED** [review]

**Pull request merged**